examples/sbol2/CreLoxRecombination.ipynb

```diff
@@ -0,0 +1,184 @@
+    "doc.addComponentDefinition(loxP1)\n",
+    "\n",
+    "# Component for the second loxP site\n",
+    "loxP2 = sbol2.ComponentDefinition('loxP_site_2', sbol2.BIOPAX_DNA)\n",
+    "loxP2_seq = sbol2.Sequence('loxP_seq_2', 'ATAACTTCGTATAATGTATGCTATACGAAGTTAT', sbol2.SBOL_ENCODING_IUPAC)\n",
+    "loxP2.sequences = [loxP2_seq.identity]\n",
+    "doc.addComponentDefinition(loxP2)\n",
+    "\n",
+    "# Create a Component for the CDS (Coding Sequence)\n",
+    "# The CDS is unknown or \"not yet selected,\" so we can use a generic placeholder sequence\n",
+    "cds = sbol2.ComponentDefinition('unknown_CDS', sbol2.BIOPAX_DNA)\n",
+    "cds_seq = sbol2.Sequence('unknown_cds_seq', 'NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN', sbol2.SBOL_ENCODING_IUPAC)\n",
+    "cds.sequences = [cds_seq.identity]\n",
+    "doc.addComponentDefinition(cds)\n",
+    "\n",
+    "# Add the loxP sites and CDS as components to the CreLox_Recombination_System ComponentDefinition\n",
+    "loxP1_comp = recombination_system.components.create('loxP1_component')\n",
+    "loxP1_comp.definition = loxP1.persistentIdentity\n",
+    "\n",
+    "cds_comp = recombination_system.components.create('cds_component')\n",
+    "cds_comp.definition = cds.persistentIdentity\n",
+    "\n",
+    "loxP2_comp = recombination_system.components.create('loxP2_component')\n",
+    "loxP2_comp.definition = loxP2.persistentIdentity\n",
+    "\n",
+    "# Create SequenceAnnotations to represent the loxP sites and the CDS\n",
+    "# Annotation for the first loxP site\n",
+    "loxP1_annotation = sbol2.SequenceAnnotation('loxP1_annotation')\n",
+    "loxP1_range = sbol2.Range('loxP1_range', 1, len(loxP1_seq.elements))\n",
+    "loxP1_annotation.locations.add(loxP1_range)\n",
+    "loxP1.sequenceAnnotations.add(loxP1_annotation)\n",
+    "\n",
+    "# GenericLocation for the CDS (unknown or not yet selected, so we leave boundaries flexible)\n",
+    "cds_annotation = sbol2.SequenceAnnotation('cds_annotation')\n",
+    "cds_location = sbol2.GenericLocation('cds_generic_location')\n",
+    "cds_location.orientation = sbol2.SBOL_ORIENTATION_REVERSE_COMPLEMENT  # Initially reverse\n",
+    "cds_annotation.locations.add(cds_location)\n",
+    "cds.sequenceAnnotations.add(cds_annotation)\n",
+    "\n",
+    "# GenericLocation for the second loxP site (second recombination site)\n",
+    "loxP2_annotation = sbol2.SequenceAnnotation('loxP2_annotation')\n",
+    "loxP2_location = sbol2.GenericLocation('loxP2_generic_location')\n",
+    "loxP2_location.orientation = sbol2.SBOL_ORIENTATION_INLINE\n",
+    "loxP2_annotation.locations.add(loxP2_location)\n",
+    "loxP2.sequenceAnnotations.add(loxP2_annotation)\n",
+    "\n",
+    "# Create SequenceConstraints to describe the relationships\n",
+    "# Constraint to ensure that loxP1 and CDS are adjacent\n",
+    "constraint1 = recombination_system.sequenceConstraints.create('loxP1_to_CDS')\n",
+    "constraint1.subject = loxP1_comp.identity\n",
```

Comment:
I don't believe you need to pull identity out on any of these. I believe the assignment routine will do that automatically for you.

Reply:
Addressed in latest commit.